esm_runscripts/slurm.py

```diff
@@ -67,9 +67,11 @@ def calc_requirements_multi_srun(self, config):
 
 
     @staticmethod
-    def mini_calc_reqs(config, model, hostfile, start_proc, end_proc):
+    def mini_calc_reqs(path, config, model, start_proc, start_core, end_proc, end_core):
```

Comment:
This will break `start_proc, end_proc = self.mini_calc_reqs(config, model, hostfile, start_proc, end_proc)` in the method `calc_requirements_multi_srun`.

I also don't like the idea of not passing a file object (e.g. `hostfile`) to the `mini_calc_reqs` method and using `with open('<file>', 'a') afile:` inside `mini_cal_reqs` instead. Note that if you pass the file object as it was done before (see the next comment) you can use `'w'` when you open the file outside of the method which rewrites whatever file already existed. If you have it inside this method and you use `'a'` for `open` the file it won't be possible to use multiple times `esm_runscripts` without cleaning the experiment directory, as this will be appending lines to an already existing file from the previous `esm_runscript` call, I would expect this to be a source of errors.